int16 axis0_lr
int16 axis0_ud
int16 axis1_lr
int16 axis1_ud
int16 axis2_lr
int16 axis2_ud
int16 dpad_lr
int16 dpad_ud
uint8 button_a
uint8 button_b
uint8 button_x
uint8 button_y
uint8 button_l1
uint8 button_l2
uint8 button_r1
uint8 button_r2

uint8 PRESSED = 0
uint8 RELEASED = 1
uint8 NOCHANGE = 2